### PR TITLE
Add `sol` alias to Solidity

### PR DESCRIPTION
**Pull request by minaminao**
## Description

Solidity source codes have `.sol` extension, and as with other languages (e.g., `rs` in Rust or `cpp` in C++), adding `sol` alias is more convenient when writing markdown.

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a syntax highlighting grammar: [URL to grammar repo]
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  - Old: [URL to grammar repo]
  - New: [URL to grammar repo]

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Comment by Alhadis** [comment]
> adding `sol` alias is more convenient when writing markdown.

Convenient, yes. Unambiguous, no. Unlike `rs` or `cpp`, the abbreviation `sol` has widespread use in contexts unrelated to Solidity; this means there's an elevated risk of Linguist supporting another language in future whose name is commonly abbreviated to "Sol". In fact, as _"Sol"_ is the Latin word for "sun", it wouldn't surprise me if a developer chooses this name for their new programming language (hey, it might be a pun on Lua…).

In other words, it's better if we remain conservative in what we associate with generic-sounding names.

**Comment by minaminao** [comment]
Indeed. I had read #4904 and understood that it is important to be unambiguous, but I did not think we should be conservative in adding an alias that is the same as the extension. One technical reason to avoid adding is that it is difficult to apply heuristics to code blocks, right? You may close this PR.

**Comment by Alhadis** [comment]
> One technical reason to avoid adding is that it is difficult to apply heuristics to code blocks, right?

Actually, heuristics _don't_ apply to code blocks: in fact, no classification is performed on highlighted code-blocks whatsoever. That's why it's important we avoid naming conflicts here, for the concerns I described earlier.
